### PR TITLE
Adding team membership and team repository support for #2 and #3

### PR DESCRIPTION
**Pull request by JacobASeverson**
Rolling team membership support into this team repository PR since
I needed some of those commits to finish this one.
- Implemented resource, added acceptance tests, and added website page for team membership
- Implemented resource, added acceptance tests, and added website page for team repositories

### CODE REVIEW

File: builtin/providers/github/resource_github_team_membership_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubTeamMembershipDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGithubTeamMembershipConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubTeamMembershipExists("github_team_membership.test_team_membership", &membership),
+					testAccCheckGithubTeamMembershipRoleState("github_team_membership.test_team_membership", &membership),
+				),
+			},
```
Comment (johnrengelman) [inline builtin/providers/github/resource_github_team_membership_test.go:32]: I think for this one we should have a 2nd `TestStep` that does an update to the role and we verify that it's recreated.

Comment (johnrengelman) [inline builtin/providers/github/resource_github_team_membership_test.go:32]: Example: https://github.com/objectpartners/terraform/blob/f-github-provider/builtin/providers/github/resource_github_team_test.go#L20-L34

File: website/source/layouts/github.erb [hunk 2]
```diff
@@ -16,6 +16,12 @@
 					<li<%= sidebar_current("docs-github-resource-membership") %>>
 					<a href="/docs/providers/github/r/membership.html">github_membership</a>
 					</li>
+					<li<%= sidebar_current("docs-github-resource-team-membership") %>>
+                    <a href="/docs/providers/github/r/team_membership.html">github_team_membership</a>
+                    </li>
```
Comment (johnrengelman) [inline website/source/layouts/github.erb]: Tabs vs spaces happening here?

Comment (JacobASeverson) [inline website/source/layouts/github.erb]: Yeah I'm not sure what's going on there, I did a spaces->tabs reformat with intellij on it and didn't seem to help but I'll try something else

File: builtin/providers/github/resource_github_team_membership_test.go (hunk 1 above)
Comment (JacobASeverson) [inline builtin/providers/github/resource_github_team_membership_test.go:32]: Sounds good - I'll  put that in

**Comment by JacobASeverson** [comment]
Added a test step to make sure both the resource and github state match what we are expecting for role.

**Pull request merged**